src/main/java/htsjdk/samtools/cram/build/CRAMReferenceRegion.java

```diff
@@ -113,17 +115,18 @@ public void fetchReferenceBases(final int referenceIndex) {
 
         // Re-resolve the reference bases if we don't have a current region or if the region we have
         // doesn't span the *entire* contig requested.
+        final SAMSequenceRecord newSequenceRecord = getSAMSequenceRecord(referenceIndex);
         if ((referenceIndex != this.referenceIndex) ||
                 regionStart != 0 ||
-                (regionLength < referenceBases.length)) {
+                (regionLength != newSequenceRecord.getSequenceLength())) {
             setCurrentSequence(referenceIndex);
             referenceBases = referenceSource.getReferenceBases(sequenceRecord, true);
```

Comment:
There is still a potential order-of-operations issue here: since the call to `getReferenceBases()` uses the instance variable `sequenceRecord`, the call to `setCurrentSequence()` must happen first. If the order of the statements were reversed in some future refactoring, we might have another terrible bug. Either add a comment explaining that `setCurrentSequence()` must happen first, or eliminate the order dependency by using `newSequenceRecord` in the `getReferenceBases()` call.